### DISCUSSION TITLE
Calculate and transfer generalised colour remapping from few known good scene reference images to larger scene set of poorly colourised target images

### DISCUSSION THREAD

**Discussion by rleaver152** [ImageMagick/ImageMagick#2334] [Help] [answered]
I have generated colour images,  auto colourised by AI from b/w originals.   The colourisation is variable and can be poor.   

For example the two files attached " TO_CORRECT PIC " are part of a large set of this scene,  not well colourised and there is an overall sepia tint.     I also have two identical images but manually colourised which represent a far better colourisation.  These are " CALIBRATION PIC " but only have the two for the scene.

I would like to  

1) get a mapping from the 'calibration" frames that computes the appropriate 'skewness' of the colour relationship between the  poor colours of the "to correct" to the "calibration" frame colours and :

2) apply  a colour correction  to all "to_correct" frames (the large set of pics) on the basis that the bulk of the colours are simply skewed (eg sepia not olive green)  rather than out and out wrong (eg purple not green grass).     Ideally form a view from a number of calibration frames from different scenes that characterise how the colourisation algorithm is going wrong (many times its consistently wrong so hope to simply remap an 'average' colour skew correction from a set of reference frames.    Maybe therefore the mapping will not include gross differences but focus on the 60:40 or 80:20 cases which are reasonable tries at correct colour but didn't quite get there

Grateful if there is a way of using ImageMagick to work out the colour 'skew' from one or more 'calibration' frames and then apply a generalised correction to a group of one or more 'to correct' frames accordingly.   It'll never be perfect, but the assumption is that the AI is consistently wrong in its colour choices in similar scenes and if we can correct from a generalised set of reference calibrated frames per scene, a broad based correction can be applied by ImageMagick.   There are then written out losslessly to output frames.

If all this is impossible,  then a great way to correct the underlying 'sepia tint' would be wonderful.

A1 CALIBRATION PIC 1
![000811_calibration](https://user-images.githubusercontent.com/19638129/88674013-2c705600-d0e1-11ea-866a-dfc12baf5721.png)

A2 CALIBRATION PIC 2
![000285_calibration](https://user-images.githubusercontent.com/19638129/88674028-32663700-d0e1-11ea-89dd-fef7c93c790c.png)

A3 TO_CORRECT PIC 1
![000285_to_correct](https://user-images.githubusercontent.com/19638129/88674032-33976400-d0e1-11ea-93ae-75c11aa41c95.png)

A4 TO_CORRECT PIC2
![000811_to_correct](https://user-images.githubusercontent.com/19638129/88674036-34c89100-d0e1-11ea-8724-7d45a4e22b34.png)

Here are the files as downloads in case this site compresses   

https://we.tl/t-HnfnNsthmO
https://we.tl/t-GlDC7ckLaQ
https://we.tl/t-rXqaswHKmo
https://we.tl/t-EthcMIKYSB

Thank you in advance for help

**Comment by snibgo** [comment, marked as answer]
The "error" in the auto-coloured versions is not a simple bias, eg with a sepia tint. It has made different decisions to the manual version. For example the kilt is a strong red colour in the hand-coloured version. To correct this, we would need to identify "pixels that are in the kilt". This is a machine-learning problem, beyond the capabilities of ImageMagick.

Ignoring that problem, we can use ImageMagick for some corrections. See [Gain and bias](http://im.snibgo.com/gainbias.htm). For example, a simple gain-and-bias:

```
call %PICTBAT%imgGainBias A4_TO_CORRECT_PIC2.png A1_CALIBRATION_PIC_1.png a4_corrected_gb.jpg
```
![a4_corrected_gb](https://user-images.githubusercontent.com/33812368/88686760-52045c00-d0ef-11ea-96a6-7a1019bf3fd8.jpg)

Another method, matching histograms:
```
call %PICTBAT%matchHisto A4_TO_CORRECT_PIC2.png A1_CALIBRATION_PIC_1.png a4_corrected_mh.jpg
```
![a4_corrected_mh](https://user-images.githubusercontent.com/33812368/88686822-65172c00-d0ef-11ea-9359-2b4d52ac827f.jpg)


With these methods, if you like the corrected version, you can appy the same transformation to any other frames.

However, in both of these methods, the desired red for Jamie's kilt has been applied to the Doctor's coat. Eek. A better result would come from a hand-coloured version that doesn't attempt major colour changes to particular objects.

**Reply by rleaver152** [reply]
Thanks I hadn't seen this before .    I need to research how to configure this for mac command line invocation.

**Reply by rleaver152** [reply]
I think Fred has it as transfercolor  for Mac also.

**Reply by rleaver152** [reply]
This method plus follow up whitebalancing seems to do what I need.   Just needs a good choice of calibration photo as kindly pointed out.  Thank you again
